<?xml version="1.0"?>

<launch>


  <!-- FCU communication -->
  <node name="rosflight_io" pkg="rosflight" type="rosflight_io" output="screen">
    <param name="udp" value="false"/>
    <param name="port" value="/dev/ttyACM0"/>
  </node>

  <!-- Motion Capture Node-->
  <node name="mocap" pkg="optitrack_vrpn" type="node" output="screen"/>

  <!-- Load common parameters -->
  <rosparam command="load" file="$(find roscopter)/params/F450.yaml"/>

  <!-- PID Position Controller -->
  <node name="controller" pkg="roscopter" type="controller">
    <remap from="estimate" to="odom"/>
    <remap from="command" to="normal_command"/>
  </node>

  <!-- Estimator -->
  <rosparam command="load" file="$(find roscopter)/params/ekf.yaml" ns="estimator"/>
  <node name="estimator" type="ekf_node" pkg="roscopter" output="screen">
    <remap from="imu" to="/imu/data"/>
    <remap from="pose" to="/MOTIONCAPTUREASSET_ned"/>
    <!-- <remap from="range" to=<RANGE_TOPIC> /> -->
  </node>

    <!--Failsafe-->
  <node name="failsafe_manager" pkg="roscopter" type="failsafe_manager.py" output="screen">
    <remap from="state" to="odom"/>
  </node>
  <node name="failsafe_controller" pkg="roscopter" type="controller" output="screen">
    <remap from='high_level_command' to="fs_high_level_command"/>
    <remap from='command' to="failsafe_command"/>
    <remap from="estimate" to="odom"/>
  </node>
  <node name="fs_multiplexer" pkg="rosflight" type="failsafe_multiplexer" output="screen">

  <!-- Waypoint Manager -->
  <node name="waypoint_manager" pkg="roscopter" type="waypoint_manager.py" output="screen">
    <remap from="waypoint" to="high_level_command"/>
    <remap from="state" to="odom"/>
  </node>

</launch>
